### PR TITLE
Add redirects array to the response

### PR DESCRIPTION
**Pull request by motoenduroboy**
Adds a redirect array allowing users to see where it goes

Image of why this is useful
Image of why this is useful

- Added redirects array

### CODE REVIEW

**Comment by motoenduroboy** [comment]
hmm it seems that github doesn't like imgur urls (well for me at least) heres the link just in case https://imgur.com/a/BK9siIF

Thanks for the quick reply!
I like your idea of the headers
maybe a array of objects?
```javascript
{
  url: "",
  response: {}
}
```
or not the whole response just the headers?

**Comment by RubenVerborgh** [comment]
> hmm it seems that github doesn't like imgur urls

You can just paste an image, and GitHub takes care of the uploading for you.

> I like your idea of the headers
> maybe a array of objects?

Makes sense.

> or not the whole response just the headers?

Just the headers, I'd say. Perhaps status code too.

**Comment by motoenduroboy** [comment]
Does this look good?
https://imgur.com/a/ZK3hZAE

**Comment by RubenVerborgh** [comment]
Why the screenshot, just text is easier to read 😄 

But yes, looks good. Now let's make that configurable, have tests, and we're good to go.

File: README.md [hunk 1]
```diff
@@ -73,6 +73,8 @@ In addition to the [standard HTTP](https://nodejs.org/api/http.html#http_http_re
 the following per-request options are supported:
 - `followRedirects` (default: `true`) – whether redirects should be followed.
 
+- `trackRedirects` (default: `false`) – whether to track redirects to a array.
```
Comment (RubenVerborgh) [inline README.md]: *an

Comment (RubenVerborgh) [reply README.md]: Also, mention the name of the array here.

File: index.js [hunk 2]
```diff
@@ -163,6 +164,13 @@ RedirectableRequest.prototype._processResponse = function (response) {
   // referenced by the Location field value,
   // even if the specific status code is not understood.
   var location = response.headers.location;
+  if (this._options.trackRedirects === true) {
+    this._redirects.push({
+      "url": this._currentUrl,
+      "headers": response.headers,
+      "statusCode": response.statusCode
+    });
```
Comment (RubenVerborgh) [inline index.js]: Please unquote all of these keys and add a dangling comma to the last. (Linter is failing.)

**Comment by RubenVerborgh** [comment]
Merged as d18d79db7555d70796796557a7227eb9cd2cbcdd.

**Pull request closed without merging**